packages/notifications/push/amplify_push_notifications/lib/src/amplify_push_notifications_impl.dart

```diff
@@ -121,14 +154,28 @@ class AmplifyPushNotifications extends PushNotificationsPluginInterface {
 
     await _registerDeviceWhenConfigure();
     _attachEventChannelListeners();
-    onNotificationReceivedInBackground(_backgroundNotificationListener);
+
+    // Explicitly set the service provider so Analytics can be recorded when notification arrives in Background/Killed state
+    _flutterApi.setProvider(_serviceProviderClient);
+
     final rawLaunchNotification = await _hostApi.getLaunchNotification();
     if (rawLaunchNotification != null) {
       final launchNotification =
           PushNotificationMessage.fromJson(rawLaunchNotification);
       _launchNotification = launchNotification;
       _recordAnalyticsForLaunchNotification(launchNotification);
     }
+
+    // Register the callback dispatcher
+    await _registerCallback(
+      _backgroundProcessor,
```

Comment:
How does this get called on iOS?

Reply:
It doesn't, I updated to do different things on Android and iOS